### ISSUE TITLE
loadStructure is extremely slow

### ISSUE THREAD

**Issue by i-sf** [nette/database#278]
Version: 3.0 (although the offending code is still in `master`).

### Bug Description

`src/Database/Structure.php:loadStructure` is *really* slow. It's performing **two queries per table**. I currently have near 100 tables, and on my local machine, each query takes 50ms. This easily adds up to 10 seconds to init! Because every request to my server spins up a new connection, and every insert loads the structure, every single request inserting things takes 10 seconds.

https://github.com/nette/database/blob/4862e4d206c7dbd63c7f113e2c70c7d154cb6538/src/Database/Structure.php#L180-L212

### Steps To Reproduce

1. Have a database with a lot of tables.
2. Have a connection to said database where each query takes roughly 50ms.
3. Profit. It's really slow.

### Expected Behavior

It should *not* be this slow. I'd expect the method to be implemented with some form of `JOIN`, to bring all the information it needs in a single query. Not to need 200 queries!

### Possible Solution

I do not know how the driver works, but rather than doing:

```
tables = fetch tables
foreach table
    fetch columns
```

I would expect:

```
tables_and_columns = fetch tables join columns
```

**Comment by MartinMystikJonas** [comment]
Well the real question here is why is loadStructure called many times. Because structure should be loaded once to cache and then cached structure should be used.

**Comment by i-sf** [comment]
Yes, I should note (but I figured this out after posting the issue; sorry about that) that using a cache storage other than memory cache indeed solves the problem (I'm not sure if I can configure our server setup in any way so that the memory cache persists across different requests to the server, or if our PHP code will inevitably create a new, clean database instance without cache every time; but that's no longer an issue with Nette). A persistent cache does indeed solve this issue for us.

I think the 50ms delay occurs because our MySQL database is hosted in an Amazon AWS cluster, so there's probably a lot of latency from our local machines to it.

Still, if possible, I think it would be nice if this method could be implemented in a more efficient way (as I suggested, with a `JOIN`, perhaps with some database-specific knowledge on which "magic tables" to look for, but I have not dug up Nette's code, so I don't even know if this is possible).

Thank you for the prompt response. I fully understand if this is considered to be a non-issue.
